src/Transport/Addressing/Partitioning/ReadOnlySettingsExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace NServiceBus.AzureServiceBus.Addressing.Partitioning
+{
+    using System.Collections.Generic;
+    using System.Configuration;
+    using System.Linq;
+    using Settings;
+    using Transport.AzureServiceBus;
+
+    public static class ReadOnlySettingsExtensions
+    {
+        public static List<RuntimeNamespaceInfo> GetPartitioningNamespaces(this ReadOnlySettings settings)
```

Comment:
I think we are looking at it from a different perspective? I wasn't thinking `transport.NamespacePartitioning().UseStrategy<T>()` needed to change.

The only change would need to be our example of how to implement a custom partitioning strategy, which would change to something like this :

```
public class ReplicatedNamespacePartitioningStrategy : INamespacePartitioningStrategy
{
    static ILog log = LogManager.GetLogger<ReplicatedNamespacePartitioningStrategy>();
    List<RuntimeNamespaceInfo> namespaces;

    public ReplicatedNamespacePartitioningStrategy(ReadOnlySettings settings)
    {
        namespaces = settings.GetPartitioningNamespaces();
        if ( namespaces.Count <= 1)
        {
           throw new Exception("The 'Replicated' namespace partitioning strategy requires more than one namespace. Configure additional connection strings");
        }
    }

    public IEnumerable<RuntimeNamespaceInfo> GetNamespaces(PartitioningIntent partitioningIntent)
    {
        log.Info($"Determining namespace for {partitioningIntent}");
        return namespaces.Select(
            selector: namespaceInfo =>
            {
                log.Info($"Choosing namespace {namespaceInfo.Alias} ({namespaceInfo.ConnectionString})");
                return namespaceInfo;
            });
    }
}
```